api/db/db_models.py

```
@@ -1984,6 +1984,113 @@ def alter_db_column_type(migrator, table_name, column_name, new_column_type):
+            if settings.DATABASE_TYPE.upper() == "POSTGRES" or is_gaussdb_compatible_database():
+                # Legacy integers were tenant_llm row ids, never 32-char tenant_model.id
+                # values. Cast them away rather than leaving strings such as "42".
+                DB.execute_sql(
+                    f'ALTER TABLE "{table_name}" ALTER COLUMN "{column_name}" '
+                    f"TYPE varchar(32) USING CAST(NULL AS varchar(32))"
+                )
+            else:
+                migrate(migrator.alter_column_type(table_name, column_name, target_field))
+                DB.execute_sql(
+                    f"UPDATE `{table_name}` SET `{column_name}` = NULL "
+                    f"WHERE `{column_name}` IS NOT NULL AND CHAR_LENGTH(`{column_name}`) <> 32"
```

Ruff (0.16.2)

[error] Possible SQL injection vector through string-based query construction

(S608)